Host/Middleware/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Core.Entities;
+using Core.Models;
+using DataAccess.Context;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Host.Middleware
+{
+    public class ExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IServiceProvider _serviceProvider;
+
+        private readonly DateTime _dateTime = DateTime.Now;
+
+        public ExceptionMiddleware(RequestDelegate next, IServiceProvider serviceProvider)
+        {
+            _next = next;
+            _serviceProvider = serviceProvider;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (DivideByZeroException ex)
+            {
+                using (var serviceScope = _serviceProvider.CreateScope())
+                {
+                    var dataContext = serviceScope.ServiceProvider.GetService<ClinicContext>();
+                    dataContext!.Add(new ExceptionEntity(ex.GetType().Name, _dateTime, ex.StackTrace??String.Empty, context.Request.Path));
```

Comment:
```suggestion
                    dataContext!.Add(new ExceptionEntity(ex.GetType().Name, _dateTime, ex.StackTrace??string.Empty, context.Request.Path));
```
If it doesn't break code, please change it to just string not String. Yesterday Andrii asked us a question about using just "", string.Empty and String.Empty, do you remember?)